library/core/src/clone.rs

```diff
@@ -38,7 +38,25 @@
 
 mod uninit;
 
-/// A common trait for the ability to explicitly duplicate an object.
+/// A common trait that allows explicit creation of a "duplicate" value.
+///
+/// ## Important Note on "Duplication"
+///
+/// What "duplication" means depends on the type implementing `Clone`:
+///
+/// - For most types, calling [`clone`] creates a semantically independent copy with its own
+///   unique identity and state.
+/// - For smart pointers (like [`Arc`], [`Rc`]), [`clone`] increases the reference count but points
+///   to the same underlying data. This means modifications to the underlying data through one
+///   clone will be visible through other clones.
+/// - For reference types (`&T`), [`clone`] just creates another reference to the same value.
```

Comment:
I don't think that the behavior actually varies much. What is duplicated is the value.

**However**, some types indeed, as values, are instead references to other data. So you get a new reference. It might be more concise to simply state that you always get a new value, it's just not guaranteed that the new value does not contain references to other data that is not duplicated.

Reply:
Besides, this part of the text is just repeated on the docs for `Clone::clone` itself.